src/data/retrieval_dataloader.py

```diff
@@ -0,0 +1,260 @@
+
+
+@data_loader_factory.register_data_loader_cls(MmtRetrievalDataConfig)
+class MmtRetrievalDataLoader(data_loader.DataLoader):
+  """A class to load dataset for mmt retrieval task."""
+
+  def __init__(self, params):
+    """Inits `MmtRetrievalDataLoader` class.
+
+    Args:
+      params: A `MmtDataConfig` object.
+
+    """
+    self._params = params
+    self._image_data_field = params.image_data_field
+    self._text_special_token_field_dict = json.loads(params.text_special_token_field_dict)
+    self._is_prebuilt_pairs = params.input_path != ''
+    self.image_name_to_features = self._image_name_to_features()
+    self.text_name_to_features = self._text_name_to_features()
+
+  def _image_name_to_features(self):
+    # TODO(roylu): Unify the name of index during preprocessing.
+    if self._is_prebuilt_pairs:
+      index_key = 'image_index'
+    else:
+      index_key = 'index'
+
+    name_to_features = {
+        index_key: tf.io.FixedLenFeature([], tf.int64),
+        self._image_data_field: tf.io.FixedLenFeature([], tf.string)
+    }
+    return name_to_features
+
+  def _text_name_to_features(self):
+    # TODO(roylu): Unify the name of index during preprocessing.
+    if self._is_prebuilt_pairs:
+      index_key = 'text_index'
+    else:
+      index_key = 'index'
+
+    name_to_features = {
+        index_key: tf.io.FixedLenFeature([], tf.int64),
+        'gt_image_index': tf.io.FixedLenFeature([], tf.int64),
+    }
+    for k in self._text_special_token_field_dict.keys():
+      name_to_features[k] = tf.io.FixedLenFeature([], tf.string,
+                                                  default_value='')
+    return name_to_features
+
+  def load(self,
+           input_context: Optional[tf.distribute.InputContext] = None):
+    """Returns a tf.dataset.Dataset."""
+
+    input_keys = ['word_ids',
+                  'segment_ids',
+                  'patch_embeddings',
+                  'att_mask',
+                  'relative_att_ids']
+
+    label_keys = ['label_ids',
+                  'label_weights']
+
+    keep_feature_keys = []
+    keep_feature_keys.extend(label_keys)
+    keep_feature_keys.extend(input_keys)
+
+    config = self._params
+
+    tokenizer = tf_text.BertTokenizer(config.vocab_filename,
+                                      lower_case=True,
+                                      preserve_unused_token=True,
+                                      token_out_type=tf.int32)
+
+    # This dataloader is currently for prediction only. However, it could be
+    # modified for training.
+    is_training = config.is_training
+    if config.include_image_text_index:
+      index_keys = ['image_index', 'text_index', 'gt_image_index']
+      input_keys.extend(index_keys)
+      keep_feature_keys.extend(index_keys)
+
+    batch_size_per_replica = input_context.get_per_replica_batch_size(
+        config.global_batch_size)
+
+    drop_remainder = config.drop_remainder
+
+    if config.input_path != '':
```

Comment:
Isn't this equivalent to just "if config.input_path:"?

Reply:
+1 that Santi's suggestion would be more idiomatic.